From 556842108e2b2578d37cdb667f747389a1dd3afe Mon Sep 17 00:00:00 2001
From: "Kevin F. Haggerty" <haggertk@lineageos.org>
Date: Fri, 9 Mar 2018 14:38:36 -0700
Subject: [PATCH] toybox: Use ISO C/clang compatible __typeof__ in minof/maxof
 macros

* typeof, sans underscores, is a GNU/GCCism
* These macros are currently only used by xzcat

Change-Id: I7273944eda64321b2dc8415ab99de044b5463e17
---
 lib/lib.h | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/lib/lib.h b/lib/lib.h
index 0b93bde..62ed197 100644
--- a/lib/lib.h
+++ b/lib/lib.h
@@ -342,8 +342,8 @@ pid_t __attribute__((returns_twice)) xvforkwrap(pid_t pid);
 // Wrapper that discards true/false "did it exit" value.
 #define NOEXIT(x) WOULD_EXIT(_noexit_res, x)
 
-#define minof(a, b) ({typeof(a) aa = (a); typeof(b) bb = (b); aa<bb ? aa : bb;})
-#define maxof(a, b) ({typeof(a) aa = (a); typeof(b) bb = (b); aa>bb ? aa : bb;})
+#define minof(a, b) ({__typeof__(a) aa = (a); __typeof__(b) bb = (b); aa<bb ? aa : bb;})
+#define maxof(a, b) ({__typeof__(a) aa = (a); __typeof__(b) bb = (b); aa>bb ? aa : bb;})
 
 // Functions in need of further review/cleanup
 #include "lib/pending.h"
-- 
2.7.4

